tools/clang/lib/SPIRV/EmitSPIRVAction.cpp

```diff
@@ -802,6 +949,17 @@ case BO_##kind : {                                                             \
       return theBuilder.getConstantFloat32(1.0);
     }
 
+    if (hlsl::IsHLSLVecType(type)) {
+      const QualType elemType = hlsl::GetHLSLVecElementType(type);
+      const uint32_t elemOneId = getValueOne(elemType);
+
+      const size_t size = hlsl::GetHLSLVecSize(type);
+      llvm::SmallVector<uint32_t, 4> elements(size, elemOneId);
```

Comment:
don't you need to handle the `vec1` corner case here? ConstantComposite shouldn't be used for vector of size 1